app/src/main/java/com/example/drawingactivity/DrawViewModel.kt

```diff
@@ -31,37 +34,41 @@ class DrawViewModel : ViewModel() {
     var onUpdate: ((ViewState) -> Unit)? = null
 
     /**
-     * Invoked when a Layer is added to the List
-     * Takes in a String and returns nothing
-     * This informs the user on the result of adding a String to the list
+     * Invoked when a String is added to the List.
+     * Takes in a String and returns nothing.
+     * This informs the user on the result of adding a String to the list.
      */
-    var onAddLayerComplete: ((String?) -> Unit)? = null
+    var onAddToList: ((String?) -> Unit)? = null
 
     /**
-     * Invoked when a Layer name is changed the List
+     * Invoked when a String is changed the List.
+     * Confirms that the String in the list has correctly Changed.
      */
-    var onEditLayerComplete: ((String?, String?) -> Unit)? = null
+    var onEditList: ((String?, String?) -> Unit)? = null
 
     /**
-     * Invoked when a Layer is removed from the List
-     * Confirms that the delete was a success
+     * Invoked when a String is removed from the List.
+     * Confirms that the delete was a success.
      */
-    var onDeleteOperationComplete: ((String?) -> Unit)? = null
+    var onDeleteFromList: ((String?) -> Unit)? = null
 
     data class ViewState(
         val primaryColor: Int,
         val currentBitmap: Bitmap?,
         val layerList: MutableList<String>,
         val currentNewLayerName: String?,
-        val editLayerIndex: Int,
-        val isDialogOpen: Boolean,
-        val isLayerSheetOpen: Boolean,
+        val listIndex: Int,
+        val targetLayer: String?,
+        val isAddDialogOpen: Boolean,
```

Comment:
You can't ever have add and edit open at the same time. Try to see if you can simplify any logic

Reply:
I was able to simplify this logic